zeek_inet_ntop.c

```diff
@@ -167,9 +172,10 @@ zeek_inet_ntop6(const u_char *src, char *dst, socklen_t size)
 			if (!zeek_inet_ntop4(src+12, tp, sizeof tmp - (tp - tmp)))
 				return (NULL);
 			tp += strlen(tp);
+			remaining -= strlen(tp);
 			break;
 		}
-		tp += sprintf(tp, "%x", words[i]);
+		tp += snprintf(tp, remaining, "%x", words[i]);
```

Comment:
Alternatively, these changes could be squashed down by replacing this with `tmp_size-(tp-tmp)`, but the pointer math felt less obvious.